packages/api/amplify_api/lib/src/method_channel_api.dart

```diff
@@ -284,55 +302,116 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
     }
   }
 
-  bool _shouldThrow(int statusCode) {
-    return statusCode < 200 || statusCode > 299;
-  }
-
-  RestResponse _formatRestResponse(Map<String, dynamic> res) {
+  AWSStreamedHttpResponse _formatRestResponse(Map<String, dynamic> res) {
     final statusCode = res['statusCode'] as int;
     final headers = res['headers'] as Map?;
-    final response = RestResponse(
-      data: res['data'] as Uint8List?,
-      headers: headers?.cast<String, String>(),
-      statusCode: statusCode,
-    );
-    if (_shouldThrow(statusCode)) {
-      throw RestException(response);
-    }
-    return response;
+    final rawResponseBody = res['data'] as Uint8List?;
+
+    return AWSStreamedHttpResponse(
+        statusCode: statusCode,
+        body: Stream.value(rawResponseBody?.toList() ?? []));
   }
 
   @override
-  RestOperation get({required RestOptions restOptions}) {
-    return _restFunctionHelper(methodName: 'get', restOptions: restOptions);
+  CancelableOperation<AWSStreamedHttpResponse> get(
+    String path, {
+    Map<String, String>? headers,
+    Map<String, dynamic>? queryParameters,
```

Comment:
Let's make these `Map<String, Object>` instead of dynamic to match `AWSHttpRequest`. There's no good reason to pass `null` as a value, for example.

Reply:
I was using dynamic here a) because that's the type used in Uri and it will convert to string as expected b) there are some cases where you might want to pass null, like if your backend implementation allowed for a null value from the client as opposed to omitted/blank string or whatever. Would prefer not to disallow people from doing something they would otherwise be allowed to do.

Would it cause an error if a Uri literal with a null value in query params passed to `AWSHttpRequest`?

Reply:
Yes, I know but my point was they allow <String, dynamic> so that you can pass <String, String> or <String, List<String>>. You're casting to <String, String> which will throw an error if they pass a <String, List<String>> or even a <String, String?> for that matter.

> Would it cause an error if a Uri literal with a null value in query params passed to AWSHttpRequest?

It wouldn't compile since type system doesn't allow it. That's the problem with `dynamic`. If you want to support null values, then make it <String, Object?>. Anything but `dynamic`.

From the Uri constructor:

```dart
Map<String, dynamic /*String|Iterable<String>*/ >? queryParameters,
```